test/integration/auth/node_test.go

```diff
@@ -248,6 +275,9 @@ func TestNodeAuthorizer(t *testing.T) {
 	expectAllowed(t, createNode2(node2Client))
 	expectAllowed(t, updateNode2Status(node2Client))
 	expectAllowed(t, deleteNode2(node2Client))
+	// node evicting pod bound to itself is allowed
+	expectAllowed(t, createNode2MirrorPod(node2Client))
+	expectAllowed(t, createNode2MirrorPodEviction(node2Client))
```

Comment:
> In this case, the removed mirror pod will be recreated.

No, in this case, we're not running a kubelet (which is what would be recreating the mirror pod based on some static pod manifest on disk)... we're simulating mirror pods ourselves manually